### PR TITLE
Managed Instance GeoRestore Cmds

### PR DESCRIPTION
**Pull request by lixiachena**
New Cmdlets for Management.Sql to allow customers to get recoverable managed databases and create database by recovery

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by hovsepm** [comment]
@lixiachena please add your spec to readme.md file as well.

**Review by jaredmoo — APPROVED** [review]

**Comment by lixiachena** [comment]
This feature is not available in public, I will add spec to readme.md once the feature is available in public.

**Comment by hovsepm** [comment]
@lixiachena please fix model validation errors:
```
error : 
operationId: RecoverableManagedDatabases_ListByInstance
scenario: List recoverable databases by managed instances
source: response
responseCode: '200'
severity: 0
errorCode: OBJECT_ADDITIONAL_PROPERTIES
errorDetails:
  code: OBJECT_ADDITIONAL_PROPERTIES
  params:
    - - location
  message: 'Additional properties not allowed: location'
  path: value/0
  title: /definitions/RecoverableManagedDatabase
  description: A recoverable managed database resource.
  similarPaths:
    - value/1
  position:
    line: 170
    column: 35
  url: >-
    specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json
  directives: {}
  jsonPosition:
    line: 12
    column: 19
  jsonUrl: >-
    specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/examples/ListRecoverableManagedDatabasesByServer.json
 error : 
operationId: RecoverableManagedDatabases_Get
scenario: Gets a recoverable databases by managed instances
source: request
responseCode: ALL
severity: 0
errorCode: REQUIRED_PARAMETER_EXAMPLE_NOT_FOUND
errorDetails:
  code: REQUIRED_PARAMETER_EXAMPLE_NOT_FOUND
  id: OAV105
  message: >-
    In operation "RecoverableManagedDatabases_Get", parameter
    recoverableDatabaseName is required in the swagger spec but is not present
    in the provided example parameter values.
 error : 
operationId: RecoverableManagedDatabases_Get
scenario: Gets a recoverable databases by managed instances
source: response
responseCode: '200'
severity: 0
errorCode: OBJECT_ADDITIONAL_PROPERTIES
errorDetails:
  code: OBJECT_ADDITIONAL_PROPERTIES
  params:
    - - location
  message: 'Additional properties not allowed: location'
  path: ''
  title: /definitions/RecoverableManagedDatabase
  description: A recoverable managed database resource.
  position:
    line: 170
    column: 35
  url: >-
    specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json
  directives: {}
  jsonUrl: >-
    specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/examples/GetRecoverableManagedDatabase.json
```

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "RecoverableManagedDatabaseProperties": {
+      "description": "The recoverable managed database's properties.",
+      "type": "object",
+      "properties": {
+        "lastAvailableBackupDate": {
```
Comment (ravbhatnagar) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json]: This should be ISO 8601 compliant and so the format should be date-time

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
```
Comment (ravbhatnagar) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json]: Is it possible to delete a recoverableDatabase through an API? Or is this a valid scenario? I am assuming that databases are automatically categorized as recoverable or not by the service.

**Review by ravbhatnagar — COMMENTED** [review]
nothing major, couple of minor notes.

**Comment by ravbhatnagar** [comment]
@lixiachena - please take a look at the comments. thanks!

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json (hunk 1 above)
Comment (lixiachena) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json]: This format is used for restore workflow . Customer don't need change any format when pass in pointintime.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json (hunk 2 above)
Comment (lixiachena) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json]: Recoverabledatabase is subtype of databases. If customer want to drop recoverable database, they can use database api to drop it. thanks

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json (hunk 1 above)
Comment (lixiachena) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json]: This format is used for restore workflow . Customer don't need change any format when pass in pointintime.

Comment (jaredmoo) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json]: Lixia, what should this property's data type be in generated C# code? string, or datetime?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json (hunk 2 above)
Comment (jaredmoo) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/recoverableManagedDatabases.json]: @ravbhatnagar their lifetime is managed by the service, so no create or delete API is possible or needed.

**Comment by hovsepm** [comment]
@ravbhatnagar per PR author your feedback is addressed. Could you verify?

**Review by KrisBash — APPROVED** [review]
Signing off for ARM

**Comment by sarangan12** [comment]
Left feedback to fix the model validator error (through IM), Waiting for response

**Review by sarangan12 — APPROVED** [review]

**Pull request merged**